source/MaterialXView/Viewer.cpp

```diff
@@ -1547,31 +1591,30 @@ bool Viewer::keyboardEvent(int key, int scancode, int action, int modifiers)
         return true;
     }
 
-    // Capture the current frame and save as an image file.
-    if (key == GLFW_KEY_F && action == GLFW_PRESS)
+    // Capture the current frame or render wedge and save as an image file.
+    if ((key == GLFW_KEY_F || key == GLFW_KEY_W) && action == GLFW_PRESS)
     {
-        _captureFilename = ng::file_dialog({ { mx::ImageLoader::TGA_EXTENSION, mx::ImageLoader::TGA_EXTENSION } }, true);
-        if (!_captureFilename.isEmpty())
+        mx::StringSet extensions = _imageHandler->supportedExtensions();
+        std::vector<std::pair<std::string, std::string>> filetypes;
+        for (const auto& extension : extensions)
+        {
+            filetypes.push_back(std::make_pair(extension, extension));
+        }
+        if (key == GLFW_KEY_F)
         {
-            if (_captureFilename.getExtension() != mx::ImageLoader::TGA_EXTENSION)
+            _captureFilename = ng::file_dialog(filetypes, true);
+            if (!_captureFilename.isEmpty())
             {
-                _captureFilename.addExtension(mx::ImageLoader::TGA_EXTENSION);
+                _captureRequested = true;
             }
-            _captureRequested = true;
         }
-    }
-
-    // Render a wedge for the current material.
-    if (key == GLFW_KEY_W && action == GLFW_PRESS)
-    {
-        _wedgeFilename = ng::file_dialog({ { mx::ImageLoader::TGA_EXTENSION, mx::ImageLoader::TGA_EXTENSION } }, true);
-        if (!_wedgeFilename.isEmpty())
+        else
         {
-            if (_wedgeFilename.getExtension() != mx::ImageLoader::TGA_EXTENSION)
+            _wedgeFilename = ng::file_dialog(filetypes, true);
```

Comment:
I like this generalization of filetypes in frame/wedge capture, but I think we still need a fallback extension for the common case where the user enters a string with no extension in the file browser (e.g. "output").  I actually ran into this myself when testing the new functionality just now.  :)

In the past I've used TGA as this default extension, as stb_image runs into edge cases with the PNG extension.

Reply:
Yes, I also ran into this. Ill restore logic to check for an extension and if none then use tga. The PNG problem I also ran into and have disallowed if using stb for unit tests. Maybe I'll disallow png here as well as it crashes.